Dl 16348 use mongo cache and not client

molafisoye opened the pull request:
# JIRA Story or Github Number - descriptive title

**Bug fix / New feature** (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## Checklist

*Reviewee* (Replace with your name)
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

*Reviewer* (Replace with your name)
 - [x]  I've confirmed that every effort has been made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've confirmed appropriate tests has been included with any code added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've confirmed code was added using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

molafisoye committed:
DL-16348 remove backlink api cache for mongo play session repo

molafisoye committed:
DL-16348 rename backlink connector

molafisoye committed:
DL-16348 use test containers

molafisoye committed:
DL-16348 ignore actual tests

molafisoye committed:
Merge remote-tracking branch 'origin/main' into DL-16348

molafisoye committed:
DL-16348 remove caching library

molafisoye committed:
DL-16348 fix tests

molafisoye committed:
DL-16348 fix integration tests

molafisoye committed:
DL-16348 rename connectors to services

molafisoye committed:
DL-16348 fix dependencies

CEPoole commented on app/repositories/SessionCacheRepository.scala:
What would be the difference between the two implementations of a `mock[SessionCacheRepository]` be? 
Gaurav is advocating for less code to maintain, which is good, but it would be good to understand why mocking would be harder.

CEPoole commented on app/connectors/DataCacheConnector.scala:
```diff
@@ -16,35 +16,25 @@
+  def fetchAndGetData[T](formId: String)(implicit hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] =
+    sessionCache.getFromSession[T](dataKey(formId))
 
-  def fetchAtedRefData[T](formId: String)(implicit hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] = {
```
Fewer changes to minimise risk of change VS cleaner code through more lines changed
There's no right answer here so you might want a quick decision with your team in stand up or something?

daglisood commented on app/connectors/BackLinkCacheService.scala:
```diff
@@ -16,45 +16,36 @@
-  val defaultSource: String = appConfig.defaultSource
-  val domain: String = appConfig.domain
+@Singleton
+class BackLinkCacheService @Inject()(
```
Then it would hold true for DataCacheConnector as well. And we should move them into service package

daglisood commented on app/connectors/DataCacheConnector.scala:
```diff
@@ -16,35 +16,25 @@
+  def saveFormData[T](formId: String, data: T)(implicit hc: HeaderCarrier, formats: Format[T]): Future[T] =
+    sessionCache.putSession(dataKey(formId), data).map(_ => data)
 
-  def fetchAndGetFormData[T](formId: String)(implicit hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] = {
```
I agree

molafisoye committed:
Merge main

Fix business logic correctness after merge

molafisoye committed:
Fix tests

molafisoye committed:
Merge main

molafisoye committed:
DL-16348 refactor file package

Matthew-George-Tyler commented on app/services/SubscriptionDataService.scala:
```diff
@@ -134,7 +132,7 @@ class SubscriptionDataService @Inject()(dataCacheConnector: DataCacheConnector,
+        case Some(x) => dataCacheService.clearCache().flatMap(r => Future.successful(r))
         case None => Future.successful(None)
       }
     } yield {
```
This for comprehension appears to swallow/mask failures

molafisoye replied:
I think we swallow the result type but not any errors - added a test for this

Matthew-George-Tyler reviewed (changes requested):
Can you investigate comments please.

molafisoye committed:
DL-16348 fix pr comments and add tests to confirm errors thrown

molafisoye committed:
DL-16348 update versions

molafisoye committed:
DL-16348 update versions

Matthew-George-Tyler reviewed (dismissed):
Acceptance tests run locally with no issues.

vierlan committed:
DL-17898: Amend test to use static dates

Amended the tests, which were failing annually at peak time because they were using dynamic current dates.

vierlan committed:
DL-17898: Update dependencies

molafisoye committed:
Merge remote-tracking branch 'origin/DL-15771' into DL-16348

# Conflicts:
#	project/AppDependencies.scala

molafisoye committed:
DL-16348 fix test

molafisoye committed:
DL-16348 fix build
